### DISCUSSION TITLE
Unsupported signature algorithm with new path // BIP-44

### DISCUSSION THREAD

**Discussion by Sotatek-ToanTran** [CoolBitX-Technology/coolwallet-js-sdk#265] [Q&A] [answered]
#Title: SOLANA // Unsupported signature algorithm with new path // BIP-44

- Description
We are trying to generate new public key from getFullPath, but we get a error like bellow:

- Expected Behavior
Expected: generated public key.

- Screenshots or appendix
![CleanShot 2022-01-27 at 11 20 01](https://user-images.githubusercontent.com/31679583/151291358-1c7560f4-2b35-4b17-be45-24f26daa86b8.png)
Input path bip44 solana

![CleanShot 2022-01-27 at 11 17 04](https://user-images.githubusercontent.com/31679583/151291194-d8c09daa-b770-428d-9d50-916c7b8c4d12.png)
File: packages/core/src/coin/derive.ts

![image (1)](https://user-images.githubusercontent.com/31679583/151291468-f931c65b-a4ff-4f02-987d-50047114a9f0.png)
Result: Unsupported signature algorithm

**Comment by hank121314** [comment, marked as answer]
`pathType` should not be `BIP44`.
`pathType` indicates the algorithm to do the key derivation. 
Not how `pathString` should be, `pathString` will always be in `BIP44` form.
